web/pandas/pdeps/0008-inplace-methods-in-pandas.md

```diff
@@ -0,0 +1,403 @@
+- Using `inplace=True` gives code that mutates the state of an object and thus has side-effects. That can introduce
+  subtle bugs and is harder to debug.
+
+Finally, there are also methods that have a `copy` keyword instead of an `inplace` keyword (which also avoids copying
+the data in the case of `copy=False`, but still returns a new object referencing the same data instead of updating the
+calling object), adding to the inconsistencies. This `copy=False` option also has become redundant with the introduction
+of Copy-on-Write.
+
+Given the above reasons, we are convinced that there is no need for neither the `inplace` nor the `copy` keyword (except
+for a small subset of methods that can actually update data inplace). Removing those keywords will give a more
+consistent and less confusing API.
+
+Thus, in this PDEP, we aim to standardize behavior across methods to make control of inplace-ness of operations
+consistent, and compatible with Copy-on-Write.
+
+Note: there are also operations (not methods) that work inplace in pandas, such as indexing (
+e.g. `df.loc[0, "col"] = val`) or inplace operators (e.g. `df += 1`). This is out of scope for this PDEP, as we focus on
+the inplace behaviour of DataFrame and Series _methods_.
+
+## Detailed description
+
+### Status Quo
+
+Many methods in pandas currently have the ability to perform an operation inplace. For example, some methods such
+as ``DataFrame.insert``, only support inplace operations, while other methods use keywords such as ``copy``
+or ``inplace`` to control whether an operation is done inplace or not.
+
+Unfortunately, many methods supporting the ``inplace`` keyword either cannot be done inplace, or make a copy as a
+consequence of the operations they perform, regardless of whether ``inplace`` is ``True`` or not. This, coupled with the
+fact that the ``inplace=True`` changes the return type of a method from a pandas object to ``None``, makes usage of
+the ``inplace`` keyword confusing and non-intuitive.
+
+In addition, some methods, such as ``DataFrame.rename`` and ``DataFrame.rename_axis`` confusingly support both
+the ``copy`` and ``inplace`` keywords, with the value of ``inplace`` overwriting the value of ``copy``.
+
+To summarize the status quo of inplace behavior of methods, we have divided methods that can operate inplace or have
+an ``inplace``/``copy`` keyword into 4 groups:
+
+**Group 1: Methods that always operate inplace**
+
+| Method Name   |
+|:--------------|
+| ``insert``    |
+| ``pop``       |
+| ``update``    |
+| ``isetitem``* |
+
+These methods always operate inplace and don't have the ``inplace`` or ``copy`` keyword.
+
+\* Although ``isetitem`` operates on the original pandas object inplace, it will not change any existing values
+inplace (it will remove the values of the column being set, and insert new values).
+
+**Group 2: Methods that modify the underlying data of the DataFrame/Series object and can be done inplace**
```

Comment:
"Methods that modify the underlying data of the DataFrame/Series object and are done inplace with `inplace=True`"